### PR TITLE
Improve error handling for remote builds

### PR DESCRIPTION
**Pull request by azazeal**
This PR now issues more accurate errors for failures during attempts to connect to remote builders. These errors should be more helpful both to users and to those monitoring exceptions. Also, each step has its own timer, with appropriate values for the step.

### CODE REVIEW

File: internal/build/imgsrc/docker.go [hunk 1]
```diff
@@ -290,57 +285,52 @@ func remoteMachine(ctx context.Context, apiClient *api.Client, appName string) (
...
+func waitForDaemon(parent context.Context, client *dockerclient.Client) (up bool, err error) {
+	ctx, cancel := context.WithTimeout(parent, 2*time.Minute)
+	defer cancel()
+
 	b := &backoff.Backoff{
 		//These are the defaults
-		Min:    200 * time.Millisecond,
-		Max:    1 * time.Second,
+		Min:    50 * time.Millisecond,
+		Max:    200 * time.Second,
```
Comment (jsierles) [inline internal/build/imgsrc/docker.go:295]: Is this intended to be 200 seconds or 200ms?

**Comment by azazeal** [comment]
ms - milliseconds. nice catch!

File: internal/build/imgsrc/docker.go [hunk 2]
```diff
@@ -211,75 +198,83 @@ func newRemoteDockerClient(ctx context.Context, apiClient *api.Client, appName s
...
+		err = fmt.Errorf("failed waiting for docker daemon: %w", err)
+		captureError(err)
+
+		return nil, err
+	case !up:
+		streams.StopProgressIndicator()
 
-		if err := waitForDaemon(errCtx, client); err != nil {
-			return errors.Wrap(err, "error waiting for docker daemon")
+		terminal.Warnf("Remote builder did not start on time. Check remote builder logs with `flyctl logs -a %s`\n", remoteBuilderAppName)
```
Comment (jsierles) [inline internal/build/imgsrc/docker.go:232]: This probably should say 'did not start in time'.

**Comment by jsierles** [comment]
I can't quite tell what the fixes were here. Could we add a brief description to the PR?

**Comment by azazeal** [comment]
Before this whole thing expired after 5m. Now it respects separate timers
and ensures we log accurate errors instead of deadline exceeded ones.

**Pull request merged**